### PR TITLE
Home work 4

### PR DESCRIPTION
**Pull request by rtmkibish**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,32 +1,27 @@
...
-  asyncStorage.run({
-    requestId: uuid(),
-    userId: uuid()
-  }, () => {
-    next();
-  });
-};
+const logger = require('./logger');
+const router = require(join(__dirname, 'routs', 'events.js'));
+const requestIdMiddleware = require(join(__dirname, 'middlewares', 'requestTracker.js'));
```
Comment (viktornord) [inline app.js]: you can require just with string, join for require is an overkill )) like for unix system it is ok

File: logger.js [hunk 2]
```diff
@@ -1,20 +1,49 @@
...
+    const state = asyncStorage.getStore().message = message;
+    logger.info(state);
+  }
+  logger.info(Object.assign(message, asyncStorage.getStore()));
+}
+
 function log(message) {
-  console.log(message);
+  if (typeof message == 'string') {
+    const state = asyncStorage.getStore().message = message;
```
Comment (viktornord) [inline logger.js]: result of assignment is a message but you should log the whole context

File: logger.js [hunk 3]
```diff
@@ -1,20 +1,49 @@
...
+    logger.log(state);
+  }
+  logger.log(Object.assign(message, asyncStorage.getStore()));
```
Comment (viktornord) [inline logger.js]: this is still executed if message is a string

File: middlewares/errorHandlers.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const { join } = require('path');
+const logger = require(join(__dirname, '..', 'logger.js'));
```
Comment (viktornord) [inline middlewares/errorHandlers.js]: why not to get context here as well

File: server1.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+              console.error(reason);
+              res.statusCode = 500;
+              res.end(JSON.stringify({error: "Unable to proccess the request"}));
+            })
+        })
+        .catch(reason => console.error(reason));
+    }]
+  ]),
+  DELETE: new Map([
+    ['/events/:id', (req, res) => {
```
Comment (viktornord) [inline server1.js:79]: would be much clear if you extract each handler to a separate file

**Review by viktornord — CHANGES_REQUESTED** [review]

File: middlewares/errorHandlers.js (hunk 4 above)
Comment (rtmkibish) [reply middlewares/errorHandlers.js]: Why do we need context here if the logging logic is in the logger file?

File: server1.js (hunk 5 above)
Comment (rtmkibish) [reply server1.js:79]: I just thought we have to do it in a single file

**Comment by rtmkibish** [comment]
Please refer to the 8a30476 commit. I've made the changes according to your comments in the 55d9c38 commit

**Review by viktornord — APPROVED** [review]

**Pull request merged**